vumi/transports/smpp/tests/test_smpp_transport.py

```diff
@@ -1524,14 +1530,20 @@ def test_delivery_report_delivered_delete_stored_remote_id(self):
 
         yield self.tx_helper.wait_for_dispatched_events(1)
 
-        self.assertFalse(
-            (yield transport.redis.exists(remote_message_key('foo'))))
+        self.assertEqual(
+            (yield transport.redis.ttl(remote_message_key('foo'))), 23)
```

Comment:
We probably want `<=` rather than `==` here, otherwise timing issues (such as a heavily loaded build machine) could cause spurious failures.